src/build.cc

```diff
@@ -1014,17 +1019,38 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
     if (content.empty())
       return true;
 
-    DepfileParser deps(config_.depfile_parser_options);
-    if (!deps.Parse(&content, err))
+    if (deps_type == "gcc") {
+      DepfileParser deps(config_.depfile_parser_options);
+      if (!deps.Parse(&content, err))
+        return false;
+
+      // XXX check depfile matches expected output.
+      deps_nodes->reserve(deps.ins_.size());
+      for (vector<StringPiece>::iterator i = deps.ins_.begin();
+           i != deps.ins_.end(); ++i) {
+        uint64_t slash_bits;
+        CanonicalizePath(const_cast<char*>(i->str_), &i->len_, &slash_bits);
+        deps_nodes->push_back(state_->GetNode(*i, slash_bits));
+      }
+    } else if (deps_type == "msvc_source_dependencies") {
+#if !NINJA_ENABLE_CL_SOURCE_DEPENDENCIES
+      *err = string("msvc_source_dependencies is only supported on Windows");
```

Comment:
nit: remove the string constructor call.

Reply:
Done.